src/lib/dnssd/platform/Dnssd.h

```diff
@@ -102,10 +102,15 @@ using DnssdResolveCallback = void (*)(void * context, DnssdService * result, con
  * The callback function SHALL NOT take the ownership of the service pointer or
  * any pointer inside this structure.
  *
+ * It is possible that this callback will be called multiple number of times.
+ * Only in case when the caller passes the CHIP_END_OF_INPUT as an error code
+ * we might assume that the mDNS browsing has ended. In that case the callback
+ * function can safely release its resources.
+ *
  * @param[in] context       The context passed to ChipDnssdBrowse or ChipDnssdResolve.
  * @param[in] services      The service list, can be nullptr.
  * @param[in] servicesSize  The size of the service list.
- * @param[in] error         The error code.
+ * @param[in] error         The error code or CHIP_END_OF_INPUT.
```

Comment:
So what's actually going on here is that any error code is treated as "last call" (that needs to be clearly documented), but CHIP_END_OF_INPUT means "last call, but process the data too"?

If so, that all needs to be clearly documented.

Reply:
I've updated the docstring. I hope that right now it will be more clear which error code is responsible for stopping browsing and which process data.